src/main/java/bio/terra/cli/command/resources/create/AiNotebook.java

```diff
@@ -0,0 +1,194 @@
+      description =
+          "Use this VM image family to find the image; the newest image in this family will be used.")
+  private String vmImageFamily;
+
+  @CommandLine.Option(
+      names = "--post-startup-script",
+      description =
+          "Path to a Bash script that automatically runs after a notebook instance fully boots up. "
+              + "The path must be a URL or Cloud Storage path, e.g. 'gs://path-to-file/file-name'")
+  private String postStartupScript;
+
+  // TODO: Add boot disk types & gpu size configs.
+
+  @CommandLine.Mixin Format formatOption;
+
+  /** Add a controlled AI Notebook instance to the workspace. */
+  @Override
+  protected void execute() {
+    createControlledResourceOptions.validateAccessOptions();
+
+    var creationParameters =
+        new GcpAiNotebookInstanceCreationParameters()
+            .instanceId(getInstanceId(globalContext.requireCurrentTerraUser()))
+            .location(location)
+            .machineType(machineType)
+            .vmImage(
+                new GcpAiNotebookInstanceVmImage()
+                    .projectId(vmImageProject)
+                    .imageFamily(vmImageFamily))
+            .postStartupScript(postStartupScript)
+            .metadata(createMetadata(workspaceContext.getWorkspaceId()));
+
+    ResourceDescription resourceCreated =
+        new WorkspaceManager(globalContext, workspaceContext)
+            .createControlledAiNotebookInstance(resourceToCreate(), creationParameters);
+    formatOption.printReturnValue(resourceCreated, AiNotebook::printText);
+  }
+
+  /** Returns a {@link ResourceDescription} for the resource to be cretaed. */
+  private ResourceDescription resourceToCreate() {
+    // build the resource object to create
+    PrivateResourceIamRoles privateResourceIamRoles = new PrivateResourceIamRoles();
+    if (createControlledResourceOptions.privateIamRoles != null
+        && !createControlledResourceOptions.privateIamRoles.isEmpty()) {
+      privateResourceIamRoles.addAll(createControlledResourceOptions.privateIamRoles);
+    }
+    return new ResourceDescription()
+        .metadata(
+            new ResourceMetadata()
+                .name(createControlledResourceOptions.name)
+                .description(createControlledResourceOptions.description)
+                .cloningInstructions(createControlledResourceOptions.cloning)
+                .controlledResourceMetadata(
+                    new ControlledResourceMetadata()
+                        .accessScope(createControlledResourceOptions.access)
+                        .privateResourceUser(
+                            new PrivateResourceUser()
+                                .userName(createControlledResourceOptions.privateUserEmail)
+                                .privateResourceIamRoles(privateResourceIamRoles))));
+  }
+
+  /** Create the metadata to put on the AI Notebook instance. */
+  private Map<String, String> createMetadata(UUID workspaceID) {
```

Comment:
So the metadata is another future parameter for the WSM API, and I think by extension to the CLI (though it's not yet in this PR). This is a nice default for users, but they might want to add more keys, or replace some of the metadata values of these keys. 
I guess it doesn't feel like default arguments are worth separating from command classes, but is that the right way to look at it? Nothing else is going to use the defaults, are they really "business logic?" The "how" of API calling feels like business logic, the "what's a good default" feels like command ergonomics. What's the separation between command classes and service classes get us here?

I think without doing defaults in the command I end up passing every option defined here (soon to be >10) to the `WorkspaceManager`, then doing Optional/null checks to create defaults and the `GcpAiNotebookInstanceCreationParameters`. If we passing an incomplete or to-be-modified `GcpAiNotebookInstanceCreationParameters` to `WorkspaceManager` for it to modify instead of the many options, that function seems like it's misleading. saying, "create it with these parameters, except in some ways that you have to look at the implementation to figure out."
If we try to separate more from the WSM API, I can define another a CLI notebook creation parameters class, but I still want its construction to handle defaults and not pass >10 parameters somewhere.

WDYT? If it is a default for a command line parameter[s], then does it stay here?

Reply:
Thanks, sounds good. I'll leave these in the command class for now. Have a TODO, but also follow up PR ready for the metadata.